### ISSUE TITLE
Unexpected Loss values during training with high learning rate

### ISSUE THREAD

**Issue by IRiViI** [keras-team/keras#13418]
<em>Please make sure that this is a Bug or a Feature Request and provide all applicable information asked by the template.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.</em>  

**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  18.04
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  '1.15.0-rc3'
- Keras version:  '2.2.4-tf'
- Python version:  3.6.8
- CUDA/cuDNN version:  I used colab
- GPU model and memory:  

**Describe the current behavior**  
I'm making plots of loss values during training and put them on top of the MSE loss surface. The values of the loss seem to be quite weird.  Maybe I'm doing something wrong or something else is going on.

**Describe the expected behavior**  
I expect the loss values to fall on the MSE surface. And this is the case when I use a low learning rate (0.001) . However, the loss value is off the MSE surface at higher loss values (0.01).

**Code to reproduce the issue**  
`
import tensorflow as tf
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import os
import time
from sklearn.preprocessing import MinMaxScaler
%matplotlib inline
import imageio

from mpl_toolkits import mplot3d

from tensorflow.keras.layers import Dense
from tensorflow.keras.models import Sequential
from tensorflow.keras.optimizers import Adam, Adagrad, Adadelta, Adamax, RMSprop, SGD

## Settings
number_of_points = 1000
total_time = 5
initial_velocity = 2 
acceleration = 3
noise = 0.5 
timepoints = np.arange(0, total_time, total_time/number_of_points)
noise = noise * np.random.randn(number_of_points)
velocities = acceleration * timepoints + noise + initial_velocity

## Create 3D graph data
def calculate_mean_square_error_loss_velocities(acceleration_tests, bias_tests):
    sum_losses = np.zeros(acceleration_tests.shape)
    for timepoint_index, timepoint in enumerate(timepoints):
        
        velocity_predict = acceleration_tests * timepoint + bias_tests
        
        sum_losses += (velocity_predict - velocities[timepoint_index])**2
        
    mean_losses=sum_losses/len(timepoints)
    return mean_losses

lowest_testing_acceleration = 0
highest_testing_acceleration = 10

lowest_bias = -5
highest_bias = 5

number_of_accelerations_to_test = 300
number_of_biases_to_test = 200

a = 3
v_0 = 2
t = timepoints
v = a * t + v_0

sample_number = 500

time_point = timepoints[sample_number]
velocity_true = velocities[sample_number]
velocity_predicted = v[sample_number]

acceleration_tests = np.linspace(lowest_testing_acceleration,highest_testing_acceleration,number_of_accelerations_to_test)
bias_tests = np.linspace(lowest_bias,highest_bias,number_of_biases_to_test)

A, B = np.meshgrid(acceleration_tests, bias_tests)

L = calculate_mean_square_error_loss_velocities(A, B)

mse_1_1 = calculate_mean_square_error_loss_velocities(np.array([1]),np.array([1]))
mse_3_2 = calculate_mean_square_error_loss_velocities(np.array([a]),np.array([v_0]))

## Create model
model = Sequential([
    Dense(1, input_shape=(1,))
])

## Model training 
learning_rate = 0.01

acceleration_prediction = 10
bias_prediction = 5

model.compile(optimizer=SGD(learning_rate), loss='mse')

weight_structure = [np.array([[acceleration_prediction]]),np.array([bias_prediction])]

model.layers[0].set_weights(weight_structure)  

number_of_itteration = 1000
number_of_samples = len(timepoints)

gd_mean_losses = []
gd_a = []
gd_b = []

for step in range(number_of_itteration):
    
    mean_loss = model.train_on_batch(timepoints, velocities)
    
    current_weights = model.layers[0].get_weights()
    acceleration_prediction = current_weights[0][0,0]
    bias_prediction = current_weights[1][0]
    
    gd_mean_losses.append(mean_loss)
    gd_a.append(acceleration_prediction)
    gd_b.append(bias_prediction)
    
## Make figure
figure = plt.figure(figsize=(14,14))
ax = plt.axes(projection='3d')
ax.plot_surface(A, B, L, rstride=1, cstride=1,
                cmap='viridis', edgecolor='none')
ax.set_title('Loss surface');
ax.set_ylabel('Initial velocity (m/s)');
ax.set_xlabel('Acceleration (m/s^2)');
ax.set_zlabel('Loss');
ax.plot(gd_a,gd_b,gd_mean_losses, linewidth=10)
ax.scatter(gd_a[::5],gd_b[::5],gd_mean_losses[::5], linewidth=10, color='blue')
sgd_data = [gd_a,gd_b,gd_mean_losses]

    
`

**Other info / logs**  
<img src="https://docs.google.com/drawings/d/e/2PACX-1vRbUvrrefcxgTA-2338Ioqjg432lUBDE2t8XYYlM-SdLtHTLrPMYtk3Z8sbrqeAmsKqNllH97WPqjzZ/pub?w=1417&amp;h=689">
